pkgs/applications/misc/hugo/rst.nix

```diff
@@ -0,0 +1,24 @@
+{ stdenv, hugo, python3, which }:
+
+stdenv.mkDerivation rec {
+  name = "hugoWithRST";
```

Comment:
Do we really need this as a separate package? All it does is adding `pygments`. Users who need hugo + pygments can just install both or write a `shell.nix` for their project.

Reply:
I tried. Installing both packages doesn't work. Hugo is able to find docutils but not pygments.
What I did was:
```
nix-env -iA nixpkgs.hugo
nix-env -iA nixpkgs.python36Packages.docutils
nix-env -iA nixpkgs.python36Packages.pygments
```
But, that resulted in hugo reporting that it couldn't find pygments.

It looks like docutils needs pygments and it is unable to find it even if both are installed (but separately) because pygments is not on docutils's PATH (or PYTHONPATH).

Additionally, sure shell.nix works but then shell customization are lost. So, I added it.

Reply:
I usually use direnv to preserve my shell: https://github.com/direnv/direnv/wiki/Nix

Reply:
It could be made a build option of hugo or a wrapper like weechat, but I would not clutter all-packages.nix with all possible dependency combinations.

Reply:
This approach is certainly simpler. I'll remove the extra package. Thank you.

Reply:
@Mic92 , I've updated the PR. Thanks.